src/test/java/io/zeebe/containers/cluster/ZeebeClusterBuilderTest.java

```diff
@@ -506,4 +508,68 @@ void shouldNotConfigureEmbeddedGateway() {
         .as("the broker is not configured to enable the embedded gateway")
         .doesNotContainEntry("ZEEBE_BROKER_GATEWAY_ENABLE", "true");
   }
+
+  @Test
+  void shouldSetImageNameForGateways() {
+    // given
+    final ZeebeClusterBuilder builder = new ZeebeClusterBuilder();
+    final String zeebeDockerImage = "camunda/zeebe:latest";
+
+    // when
+    final DockerImageName gatewayImageName = DockerImageName.parse(zeebeDockerImage);
+    builder.withGatewayImage(gatewayImageName).withGatewaysCount(1).withEmbeddedGateway(false);
+    final ZeebeCluster cluster = builder.build();
+
+    // then
+    final Map<String, ZeebeGatewayNode<? extends GenericContainer<?>>> gateways =
+        cluster.getGateways();
+    assertThat(gateways).hasSize(1).hasValueSatisfying(zeebeImageHasImageName(zeebeDockerImage));
```

Comment:
This gives an error message where we compare the exact object we want to compare, and also give a bit more context when it fails. It can also be applied to the same assertions below. One caveat, I didn't test it, so maybe it doesn't compile as is and you need to fix something, but it should more or less work :sweat_smile: 

```suggestion
    assertThat(gateways)
        .as("the only gateway created has the right docker image")
        .singleElement()
        .satisfies(gateway -> assertThat(gateway.getDockerImageName()).isEqualTo(zeebeDockerImage));
```

Reply:
Done